Update versions + cleanup

dano0b opened the pull request:
update:
docker 19.03
etcd 3.4.1
cni 0.8.2
crictl 1.16.0
kubernetes 1.15.4
calico 3.9

cleanup:
more reliable docker download
dashboard re-apply of serviceaccount removed (previous dashboard token sourced; required pod restart)

galthaus commented on krib/templates/krib-dashboard.sh.tmpl:
```diff
@@ -43,17 +43,7 @@ if [[ $MASTER_INDEX == 0 ]] ; then
   mkdir -p /root/setup/dashboard
   cp /tmp/kubernetes-dashboard.yaml /root/setup/dashboard/kubernetes-dashboard.yaml
 
-  cat > /root/setup/dashboard/dashboard-serviceaccount.yaml << EOFSA
```
@dano0b - why was this removed?  @funkypenguin may be using this.  Should it be conditional?

funkypenguin replied:
I can confirm I'm not using this - I do my dashboards using helm charts outside of DRP, driven by ansible. Whether it should be conditional or not, I have no comment on :)

dano0b replied:
@galthaus was added by me and was never needed, the same serviceaccount definition is in the `recommended.yaml` which gets downloaded. Only the clusterRoleBinding is required.

galthaus commented on krib/templates/docker-install.sh.tmpl:
```diff
@@ -4,6 +4,15 @@ set -e
 # Get access and who we are.
 {{template "setup.tmpl" .}}
+
+{{ if .ParamExists "krib/cluster-profile" -}}
```
Is the cluster profile needed for the download functions?

I bet it is.  Hmm - need to think about this.  I'd like to keep docker-install not directly requiring the krib-profile if possible.

dano0b replied:
@galthaus For now it is required since I use some functions(`download` & `with_backoff`)  of the `krib-lib.sh` which has `[[ -z "$CLUSTER_PROFILE" ]] && xiterr 1 "required CLUSTER_PROFILE not set"`. 
This functions are super generic and could be placed anywhere since they don't have any krib dependency. Any idea where to put them?

galthaus commented:
@dano0b, I added a couple of comments to think about.

dano0b commented:
> @dano0b, I added a couple of comments to think about.

@galthaus I moved the functions to the task-library and remove the dependency for the docker install on the krib lib. Test provisioning was successful.
